code/modules/mob/living/carbon/human/species/yautja/_species.dm

```diff
@@ -220,3 +221,97 @@
 
 /datum/species/yautja/handle_paygrades()
 	return ""
+
+/// Open the Yautja emote panel, which allows them to use their emotes easier.
+/datum/species/yautja/open_emote_panel()
+	var/datum/yautja_emote_panel/ui = new(usr)
+	ui.ui_interact(usr)
+
+/datum/action/yautja_emote_panel
+	name = "Open Emote Panel"
+	action_icon_state = "looc_toggle"
+
+/datum/action/yautja_emote_panel/can_use_action()
+	. = ..()
+	if(!.)
+		return FALSE
+
+	if(!isyautja(owner))
+		return FALSE
+
+	return TRUE
+
+/datum/action/yautja_emote_panel/action_activate()
+	if(!can_use_action())
+		return
+
+	var/mob/living/carbon/human/human_owner = owner
+	var/datum/species/yautja/yautja_species = human_owner.species
+	yautja_species.open_emote_panel()
+
+/datum/yautja_emote_panel
+	/// Static dict ("category" : (emotes)) of every yautja emote typepath
+	var/static/list/yautja_emotes
+	/// Static list of categories
+	var/static/list/yautja_categories = list()
+
+/datum/yautja_emote_panel/New()
+	if(!length(yautja_emotes))
+		var/list/emotes_to_add = list()
+		for(var/datum/emote/living/carbon/human/yautja/emote as anything in subtypesof(/datum/emote/living/carbon/human/yautja))
+			if(!initial(emote.key))
+				continue
+
+			if(!(initial(emote.category) in yautja_categories))
+				yautja_categories += initial(emote.category)
+			emotes_to_add += emote
+		/// I hate this method, but anything else I tried just seemed to make the whole UI blank.
+		emotes_to_add -= /datum/emote/living/carbon/human/yautja/species_sound/loudroar
```

Comment:
why doesn't checking for `!initial(emote.no_panel)` work in the for loop on 261?

Reply:
I have no idea. But it just makes the entire panel blank.